includes/class-wc-payments-upe-checkout.php

```diff
@@ -0,0 +1,299 @@
+<?php
+/**
+ * Class WC_Payments_Checkout
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay;
+
+use Exception;
+use WC_Payments;
+use WC_Payments_Account;
+use WC_Payments_Customer_Service;
+use WC_Payments_Utils;
+use WCPay\Fraud_Prevention\Fraud_Prevention_Service;
+use WCPay\Payment_Methods\UPE_Payment_Gateway;
+use WCPay\Platform_Checkout\Platform_Checkout_Utilities;
+
+
+/**
+ * WC_Payments_Checkout
+ */
+class WC_Payments_UPE_Checkout extends WC_Payments_Checkout {
+
+	/**
+	 * WC Payments Gateway.
+	 *
+	 * @var UPE_Payment_Gateway
+	 */
+	protected $gateway;
+
+	/**
+	 * Platform Checkout Utilities.
+	 *
+	 * @var Platform_Checkout_Utilities
+	 */
+	protected $platform_checkout_util;
+
+	/**
+	 * WC Payments Account.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	protected $account;
+
+	/**
+	 * WC Payments Customer Service
+	 *
+	 * @var WC_Payments_Customer_Service
+	 */
+	protected $customer_service;
+
+	/**
+	 * List of the available payment gateways.
+	 *
+	 * @var array $gateway_map
+	 */
+	protected $gateway_map = [];
+
+	/**
+	 * Construct.
+	 *
+	 * @param UPE_Payment_Gateway          $gateway                WC Payment Gateway.
+	 * @param Platform_Checkout_Utilities  $platform_checkout_util Platform Checkout Utilities.
+	 * @param WC_Payments_Account          $account                WC Payments Account.
+	 * @param WC_Payments_Customer_Service $customer_service       WC Payments Customer Service.
+	 * @param array                        $gateway_map            WC Payment Gateway list.
+	 */
+	public function __construct(
+		UPE_Payment_Gateway $gateway,
+		Platform_Checkout_Utilities $platform_checkout_util,
+		WC_Payments_Account $account,
+		WC_Payments_Customer_Service $customer_service,
+		array $gateway_map
+	) {
+		$this->gateway                = $gateway;
+		$this->platform_checkout_util = $platform_checkout_util;
+		$this->account                = $account;
+		$this->customer_service       = $customer_service;
+		$this->gateway_map            = $gateway_map;
+
+		add_action( 'wc_payments_set_gateway', [ $this, 'set_gateway' ] );
+		add_action( 'wc_payments_add_upe_payment_fields', [ $this, 'payment_fields' ] );
+	}
+
+	/**
+	 * Generates the configuration values, needed for payment fields.
+	 *
+	 * Isolated as a separate method in order to be available both
+	 * during the classic checkout, as well as the checkout block.
+	 *
+	 * @return array
+	 */
+	public function get_payment_fields_js_config() {
+
+		$payment_fields                             = parent::get_payment_fields_js_config();
+		$payment_fields['accountDescriptor']        = $this->gateway->get_account_statement_descriptor();
+		$payment_fields['addPaymentReturnURL']      = wc_get_account_endpoint_url( 'payment-methods' );
+		$payment_fields['gatewayId']                = UPE_Payment_Gateway::GATEWAY_ID;
+		$payment_fields['isCheckout']               = is_checkout();
+		$payment_fields['checkoutTitle']            = $this->gateway->title;
```

Comment:
```suggestion
```
I was wondering how this would work, because surely this would simply return the card gateway's title every time, instead of the correct gateway title for each specific gateway...but then when searching for the key `checkoutTitle`, I couldn't actually find this used anywhere in the code anymore. 

I'm not sure when this property got disappeared, but I think we can safely remove it, since it's both irrelevant and now inaccurate anyways.

Reply:
Right, it looks like `checkoutTitle` has been with us for a more extended period. I've removed the field and can confirm that nothing is broken afterward.